### PR TITLE
Avoid * select and add type casting for results

### PR DESCRIPTION
**Pull request by pwirth**
https://github.com/inblockio/mediawiki-extensions-Aqua/issues/311

### CODE REVIEW

File: includes/RevisionXmlBuilder.php [hunk 1]
```diff
@@ -51,13 +51,13 @@ public function getPageMetadataByRevId( int $revId ): string {
 		}
 
 		$output = [
-			'domain_id' => $row->domain_id,
+			'domain_id' => (int)$row->domain_id,
```
Comment (rht) [inline includes/RevisionXmlBuilder.php:54]: The domain ID is hexadecimal, and so needs to be string.

File: includes/Verification/VerificationLookup.php [hunk 2]
```diff
@@ -75,7 +75,25 @@ public function verificationEntityFromTitle( Title $title ): ?VerificationEntity
 	public function verificationEntityFromQuery( array $query ): ?VerificationEntity {
 		$res = $this->lb->getConnection( DB_REPLICA )->selectRow(
 			static::TABLE,
-			[ '*' ],
+			[
```
Comment (rht) [inline includes/Verification/VerificationLookup.php:78]: I think we need to add a comment in `sql/data_accounting.sql` to remind anyone who add a column to also add it here.

Comment (rht) [reply includes/Verification/VerificationLookup.php:78]: And also say it for WitnessLookup.php.

File: includes/Verification/WitnessLookup.php [hunk 3]
```diff
@@ -37,8 +50,10 @@ public function witnessEventFromQuery( array $query ): ?WitnessEventEntity {
 		if ( !$row ) {
 			return null;
 		}
+		$data = (array)$row;
+		$data['witness_event_id'] = (int)$data['witness_event_id'];
 
-		return new WitnessEventEntity( $row );
+		return new WitnessEventEntity( (object)$data );
```
Comment (rht) [inline includes/Verification/WitnessLookup.php:56]: Why the casting to `object` when it is already `array`?

File: includes/RevisionXmlBuilder.php (hunk 1 above)
Comment (pwirth) [reply includes/RevisionXmlBuilder.php]: fixed

File: sql/data_accounting.sql [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 -- Page verification table
 CREATE TABLE IF NOT EXISTS /*$wgDBprefix*/revision_verification (
+	-- Add new fields to DataAccounting\Verification\VerificationLookup
```
Comment (rht) [inline sql/data_accounting.sql:7]: Not explicit enough. Should be something like "Attention: if you add new columns to this table, make sure to also update the select in path/to/VerificationLookup.php inside the method xxx".

Comment (pwirth) [reply sql/data_accounting.sql]: i updated the doc

**Pull request merged**